pexec/managed_process_test.go

```diff
@@ -581,6 +582,38 @@ done`, tempFile.Name()))
 	})
 }
 
+func TestManagedProcessEnvironmentVariables(t *testing.T) {
+	t.Run("set an environment variable", func(t *testing.T) {
+		logger := golog.NewTestLogger(t)
+		output := new(bytes.Buffer)
+		proc := NewManagedProcess(ProcessConfig{
+			Name:                 "bash",
+			Args:                 []string{"-c", "printf %s $VIAM_HOME"},
+			EnvironmentVariables: map[string]string{"VIAM_HOME": "/opt/viam"},
+			OneShot:              true,
+			LogWriter:            output,
+		}, logger)
+		test.That(t, proc.Start(context.Background()), test.ShouldBeNil)
+		test.That(t, output.String(), test.ShouldEqual, "/opt/viam")
+	})
+
+	t.Run("overwrite an environment variable", func(t *testing.T) {
+		logger := golog.NewTestLogger(t)
+		// test that the variable already exists
+		test.That(t, os.Getenv("HOME"), test.ShouldNotBeEmpty)
+		output := new(bytes.Buffer)
+		proc := NewManagedProcess(ProcessConfig{
+			Name:                 "bash",
+			Args:                 []string{"-c", "printf %s $HOME"},
+			EnvironmentVariables: map[string]string{"HOME": "/some/dir"},
+			OneShot:              true,
```

Comment:
```suggestion
```
Could take this out to test the other code path.

Reply:
Good idea. Added a new test for the other code path to be explicit (rather than overloading the test.)